src/snowflake/snowpark/modin/plugin/_internal/index.py

```diff
@@ -112,7 +130,9 @@ def to_pandas(self) -> native_pd.Index:
         pandas Index
             A native pandas Index representation of self
         """
-        return self._index
+        if self.is_lazy:
+            return self._query_compiler.pandas_index()
```

Comment:
can we change to 
```
df_index_only = self._query_compiler.to_pandas(index_only=True)
return df_index_only.index
```

Reply:
Or you can just call `self._query_compiler.index`. I can help to make the refactor later.

Reply:
do i need to add index_only as a parameter and write the code path?

Reply:
NO, for now, you can just call `return self._query_compiler.index`

Reply:
that returns a modin index

Reply:
```    @property
    def index(self) -> pd.Index:
        """
        Get pandas index. The method eagerly pulls the values from Snowflake because index requires the values to be
        filled

        Returns:
            The index (row labels) of the DataFrame.
        """
        return self._modin_frame.index_columns_index
```

Reply:
there is a bug `def index(self) -> pandas.Index:`

Reply:
renamed this correctly now